scripts/docgen/generate-docs.js

```diff
@@ -240,6 +243,63 @@ function fixAllLinks(htmlFiles) {
   return Promise.all(writePromises);
 }
 
+/**
+ * Generate an temporary abridged version of index.d.ts used to create
+ * Node docs.
+ */
+async function generateNodeSource() {
+  const sourceText = await fs.readFile(sourceFile, 'utf8');
+
+  // Parse index.d.ts. A dummy filename is required but it doesn't create a
+  // file.
+  let typescriptSourceFile = typescript.createSourceFile(
+    'temp.d.ts',
+    sourceText,
+    typescript.ScriptTarget.ES2015,
+    /*setParentNodes */ false
+  );
+
+  // Traverse AST to get blocks annotated with @webonly and store their
+  // start/end index.
+  const webOnlyBlocks = [];
+  function findWebOnlyBlocks(node) {
+    if (node.jsDoc) {
+      node.jsDoc.forEach(item => {
+        if (item.tags) {
+          item.tags.forEach(tag => {
+            if (tag.tagName.escapedText === 'webonly') {
+              webOnlyBlocks.push({ start: node.pos, end: node.end });
```

Comment:
Is it possible to create a new AST by deleting "webonly" nodes? Then you can use AST.writefile or something similar to create a file instead of scanning the file and delete lines.

Reply:
The native Typescript parser is pretty basic.  It doesn't have a delete method and doesn't give direct access to children.  There is a wrapper library that provides that functionality (https://github.com/dsherret/ts-morph) but I think it also does it by just removing text?  I would use that library but it says prominently in several places it's WIP and I don't know how stable it is.

Reply:
Thanks for the help, updated the code to use the transformer as you suggested.